packages/lib-classifier/src/store/FeedbackStore.spec.js

```diff
@@ -1,150 +1,196 @@
-            successEnabled: true,
-            successMessage: 'Yay! 1-1',
-            failureEnabled: true,
-            failureMessage: 'No!'
-          },
-          {
-            id: 'testRule1-2',
-            hideSubjectViewer: false,
-            answer: '1',
-            strategy: 'testStrategy',
-            success: false,
-            successEnabled: true,
-            successMessage: 'Yay! 1-2',
-            failureEnabled: false
-          }
-        ],
-        T1: [{
-          id: 'testRule2-1',
-          hideSubjectViewer: true,
-          answer: '0',
-          strategy: 'testStrategy',
-          success: false,
-          successEnabled: true,
-          successMessage: 'Yippee!',
-          failureEnabled: true,
-          failureMessage: 'Nope!'
-        }]
-      },
-      showModal: false
-    }
-    feedback = FeedbackStore.create(feedbackStub)
-  })
-
-  it('should exist', function () {
-    expect(FeedbackStore).to.be.ok()
-    expect(FeedbackStore).to.be.an('object')
+  
+  describe('existance', function () {
+    it('should exist', function () {
+      expect(FeedbackStore).to.be.ok()
+      expect(FeedbackStore).to.be.an('object')
+    })
   })
 
-  describe('createRules', function () {
-    const project = {
-      id: '1'
-    }
-    const workflow = {
-      id: '2'
-    }
-    const subject = {
-      id: '3'
-    }
+  describe('Actions > createRules', function () {
+    let workflow, project, subject, feedback, feedbackStub
+    before(function () {
+      workflow = WorkflowFactory.build()
+      project = ProjectFactory.build({}, { activeWorkflowId: workflow.id })
+      subject = Factory.build('subject')
+      feedbackStub = FeedbackFactory.build()
+      feedback = FeedbackStore.create(feedbackStub)
+      feedback.projects = ProjectStore.create()
+      feedback.workflows = WorkflowStore.create()
+      feedback.subjects = SubjectStore.create()
+    })
 
     beforeEach(function () {
-      feedback.projects = {
-        active: project
-      }
-      feedback.workflows = {
-        active: workflow
-      }
-      feedback.createRules(subject)
+      feedback.projects.setResource(project)
+      feedback.projects.setActive(project.id)
+      feedback.workflows.setResource(workflow)
+      feedback.workflows.setActive(workflow.id)
     })
 
     afterEach(function () {
+      feedback.reset()
       helpers.isFeedbackActive.resetHistory()
       helpers.generateRules.resetHistory()
     })
 
     it('should set active state', function () {
-      expect(helpers.isFeedbackActive).to.have.been.calledOnceWith(project, subject, workflow)
+      expect(feedback.isActive).to.be.false()
+      feedback.createRules(subject)
+      expect(helpers.isFeedbackActive).to.have.been.calledOnce()
+      // Normally we would use sinon's withArgs call to test the spy for specific arguments
+      // But this is complicated with the fact that the arguments actually used in the store are MST references
```

Comment:
Can we use the MST references eg. `feedback.projects.active`? Seems clearer than using `.args[0][0]` to get the first argument of the first call.

Reply:
Does this mean to do this instead?
`expect(feedback.projects.active.toJSON()).to.deep.equal(project)`
We wouldn't be testing the spy anymore though doing this? We would be testing that the store is setting the resource map and active reference to our mocked resource. Unless I'm misunderstanding what the recommendation is here.

Reply:
I was thinking `.withArgs(feedback.projects.active…)` to check for the MST references.

Reply:
Ah yes. I tried that but now I'm not remembering why that didn't work. I'll try it again.

Reply:
It's not a blocker, and I'm not massively familiar with how feedback works internally but that just came to mind when I was reading the code.

Reply:
I can't remember what I was doing to make that not work, but I have it working now so I'll push an update since it is more readable.